compiler/rustc_const_eval/src/interpret/cast.rs

```diff
@@ -30,9 +30,22 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             }
 
             Misc => {
-                let src = self.read_immediate(src)?;
-                let res = self.misc_cast(&src, cast_ty)?;
-                self.write_immediate(res, dest)?;
+                // We have to special-case enums since they can be non-scalar if they have `repr(align)`.
+                // See issues #92464, #96185.
+                if src.layout.ty.is_enum() {
```

Comment:
cg_clif has pretty much the same special case: https://github.com/bjorn3/rustc_codegen_cranelift/blob/25f7711fea4f83a82d54c17dff5cc5b9a2b2937b/src/base.rs#L630-L652

Reply:
Ah, so clif already uses the thing I tried to do in SSA codegen -- looks like it supports getting the discriminant from an operand.

That case might become redundant if we can move this to MIR. :)